app/controllers/users_controller.rb

```diff
@@ -1,12 +1,22 @@
 class UsersController < ApplicationController
   def dashboard
-    if current_user
-      github_service = GithubService.new(current_user.github_access_token)
-      @repositories = github_service.user_repositories_with_star_count
-      @repositories = @repositories.sort_by { |repo| repo['updated_at'] }.reverse
-    else
-      redirect_to login_path, alert: 'Please log in to view your dashboard.'
-    end
+    return redirect_to login_path, alert: 'Please log in to view your dashboard.' unless current_user
```

Comment:
to moglaby byc before_action ktora jerst zdefiniowana w applicationcontroller

Reply:
apropo tego screencasta to powiedz mi czy o to chodziło :)